File: python_refactored_waddle/num_identical_pairs.py
from itertools import combinations


def num_identical_pairs(nums):
    """
    source:https://leetcode.com/problems/number-of-good-pairs/
    """
    pairs = 0
    for _ in (num for num in combinations(nums, 2) if num[0] == num[1]):
        pairs += 1
    return pairs


print(num_identical_pairs((1, 2, 3, 1, 1, 3)))

File: src/numIdenticalPairs.js
/**
 * @copyright https://leetcode.com/problems/number-of-good-pairs/
 * @param {number[]} nums
 * @return {number}
 */ export default function numIdenticalPairs(nums) {
  let target = 0;
  for (let i = 0; i < nums.length; i++) {
    for (let j = 0; j < nums.length; j++) {
      if (i < j) {
        target += nums[i] === nums[j];
      }
    }
  }
  return target;
}
console.log(numIdenticalPairs([1, 2, 3, 1, 1, 3]));
